src/Api/Endpoints/ImportNotifications/EndpointRouteBuilderExtensions.cs

```diff
@@ -75,11 +82,18 @@ private static async Task<IResult> Get(
         [RegularExpression($"^{Regexes.ChedReferenceNumber}$")]
             string chedReferenceNumber,
         [FromServices] IBtmsService btmsService,
+        HttpContext httpContext,
         CancellationToken cancellationToken
     )
     {
         var notification = await btmsService.GetImportNotification(chedReferenceNumber, cancellationToken);
+        var bcp = notification?.PartOne?.PointOfEntry;
+
+        if (bcp is null)
+            return Results.NotFound();
 
-        return notification is not null ? Results.Ok(notification) : Results.NotFound();
+        return !BcpAccessAuthorisation.ClientHasAccessTo(httpContext.User, [bcp])
```

Comment:
Nit - An extension method on the `.User` type might make this line simpler to read instead of a pure static function.